bigchaindb/core.py

```diff
@@ -390,6 +395,13 @@ def get_spent(self, txid, output):
                     ' with the chain'.format(txid))
             # if its not and invalid transaction
             if status is not None:
+                if 'metadata' not in transaction:
```

Comment:
Isn't this already handled by `Transaction.from_db` in `Bigchain.get_transaction`?

Reply:
@r-marques `transactions` here are fetched using `backend.query.get_spent`. So, they don't have metadata preloaded. We could assign transaction to a variable here `_, status = self.get_transaction ...` rather than ignoring which would help us avoid the `if`?

Reply:
I think it would be better to use the transaction returned by `get_transaction`. This way we reduce the number of database calls and we eliminate the need to handle the metadata.